src/Common/src/Interop/Windows/kernel32/Interop.GetFileAttributesEx.cs

```diff
@@ -68,10 +67,13 @@ internal unsafe struct WIN32_FIND_DATA
             internal uint nFileSizeLow;
             internal uint dwReserved0;
             internal uint dwReserved1;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 260)]
-            internal string cFileName;
-            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = 14)]
-            internal string cAlternateFileName;
+            private fixed char _cFileName[260];
+            private fixed char _cAlternateFileName[14];
+
+            internal Span<char> cFileName
+            {
+                get { fixed (char* c = _cFileName) return new Span<char>(c, 260); }
+            }
```

Comment:
I am happy with the Span worker methods.

The less than ideal part are the fixed buffer to Span conversions. The Span getters have less ceremony (lines of code) and you have to be careful how to use them. The struct wrappers would add more ceremony (lines) and you need to be careful about how to use them in different way, e.g. you need to be careful to not pass the struct by value.

I prefer less ceremony until it proves to be a problem.